### PR TITLE
Preview SDK for Microsoft.ResourceGraph

### PR DESCRIPTION
**Pull request by ilidemi**
Microsoft.ResourceGraph is a new resource provider that allows customers to explore their Azure resources at scale using a powerful query language (KQL). We are launching in Public Preview at Ignite and aiming to release CLI/PowerShell tools by then.

More info here: https://aka.ms/resource-graph/intro

Swagger spec PRs: [#3716](https://github.com/Azure/azure-rest-api-specs/pull/3716) and [#3796](https://github.com/Azure/azure-rest-api-specs/pull/3796) 
Autorest config PR: [#3795](https://github.com/Azure/azure-rest-api-specs/pull/3795)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ResourceGraph/Management/Management.ResourceGraph/Microsoft.Azure.Management.ResourceGraph.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code below this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.ResourceGraph</PackageId>
+    <Description>Provides Resource Graph operations for Microsoft Azure.</Description>
+    <AssemblyName>Microsoft.Azure.Management.ResourceGraph</AssemblyName>
+    <Version>1.0.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/ResourceGraph/Management/Management.ResourceGraph/Microsoft.Azure.Management.ResourceGraph.csproj]: Please start versioning from `0.9.0-preview`

File: src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/Microsoft.Azure.Management.ResourceGraph.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <Description>ResourceGraph.Tests Class Library</Description>
+    <Authors>Microsoft Corporation</Authors>
+    <AssemblyName>Microsoft.Azure.Management.ResourceGraph.Tests</AssemblyName>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Remove="SessionRecords\ResourceGraph.Tests.ScenarioTests.ScenarioTests\**" />
+    <EmbeddedResource Remove="SessionRecords\ResourceGraph.Tests.ScenarioTests.ScenarioTests\**" />
+    <None Remove="SessionRecords\ResourceGraph.Tests.ScenarioTests.ScenarioTests\**" />
+  </ItemGroup>
+  <!--<PropertyGroup>
```
Comment (dsgouda) [inline src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/Microsoft.Azure.Management.ResourceGraph.Tests.csproj]: Please delete commented lines

File: src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/ResourceGraphTests.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+                Assert.Equal(743, queryResponse.TotalRecords);
+                Assert.NotNull(queryResponse.SkipToken);
+                Assert.Equal(ResultTruncated.False, queryResponse.ResultTruncated);
+                Assert.NotNull(queryResponse.Data);
+                Assert.NotNull(queryResponse.Facets);
+                Assert.Equal(0, queryResponse.Facets.Count);
+
+                // Data columns
+                Assert.NotNull(queryResponse.Data.Columns);
+                Assert.Equal(1, queryResponse.Data.Columns.Count);
```
Comment (dsgouda) [inline src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/ResourceGraphTests.cs]: Please address Warnings regarding asserts

File: src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/Microsoft.Azure.Management.ResourceGraph.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,32 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.ResourceGraph.Tests</PackageId>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/Microsoft.Azure.Management.ResourceGraph.Tests.csproj]: Please replace this with `<Version>1.0.0</Version>`

File: src/SDKs/_metadata/resourcegraph_resource-manager.txt [hunk 5]
```diff
@@ -0,0 +1,29 @@
+﻿Installing AutoRest version: latest
+AutoRest installed successfully.
+Commencing code generation
+Generating CSharp code
+Executing AutoRest command
+cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resourcegraph/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=C:\git\azure-sdk-for-net\src\SDKs
+Error: Cannot write data to 'https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resourcegraph/resource-manager/generated/ResourceGraphClient.cs'. Protocol 'https:' not supported for writing. Make sure to specify a local, absolute path as target file/folder.
+
+System.Management.Automation.RuntimeException: Cannot convert value to type System.String. ---> System.Management.Automation.PSInvalidCastException: Cannot convert value to type System.String. ---> System.InvalidOperationException: Process has exited, so the requested information is not available.
```
Comment (dsgouda) [inline src/SDKs/_metadata/resourcegraph_resource-manager.txt]: Looks like code generation failed. How was the code generated?

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/Microsoft.Azure.Management.ResourceGraph.Tests.csproj [hunk 6]
```diff
@@ -2,7 +2,7 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.ResourceGraph.Tests</PackageId>
-    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/ResourceGraph/Management/ResourceGraph.Tests/Microsoft.Azure.Management.ResourceGraph.Tests.csproj]: Replace `VersionPrefix` with `Version`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from a minor comment. Thanks for addressing all the warnings

File: src/SDKs/ResourceGraph/Management/Management.ResourceGraph/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+using System.Runtime.InteropServices;
+
+[assembly: AssemblyTitle("Microsoft Azure Resource Graph Library")]
+[assembly: AssemblyDescription("Provides Microsoft Azure Resource Graph operations.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (dsgouda) [inline src/SDKs/ResourceGraph/Management/Management.ResourceGraph/Properties/AssemblyInfo.cs]: AssemblyVersion must match with that in csproj

**Review by dsgouda — CHANGES_REQUESTED** [review]
Missed a comment

**Comment by dsgouda** [comment]
@demyanenko squash commits into a single commit and we are good to merge

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by demyanenko** [comment]
@dsgouda Squashed and pushed.

**Pull request merged**